src/common/inference/mod.rs

```diff
@@ -6,3 +12,41 @@ pub mod query_requests_grpc;
 pub mod query_requests_rest;
 pub mod service;
 pub mod update_requests;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct InferenceToken(pub Option<String>);
```

Comment:
Why did you decide for this design instead of having an `Option<InferenceToken>` instead?

Reply:
We discussed with @generall that it could be an option. The reason is to simplify the code. If it's not the best approach, share your ideas to make it better.